testing/test_collection.py

```diff
@@ -1171,13 +1171,13 @@ def test_collect_symlink_file_arg(testdir):
     real = testdir.makepyfile(
         real="""
         def test_nodeid(request):
-            assert request.node.nodeid == "real.py::test_nodeid"
+            assert request.node.nodeid == "symlink.py::test_nodeid"
         """
     )
     symlink = testdir.tmpdir.join("symlink.py")
     symlink.mksymlinkto(real)
     result = testdir.runpytest("-v", symlink)
-    result.stdout.fnmatch_lines(["real.py::test_nodeid PASSED*", "*1 passed in*"])
+    result.stdout.fnmatch_lines(["symlink.py::test_nodeid PASSED*", "*1 passed in*"])
```

Comment:
i fear that this detail change may cause some troublr wrt resolution of symlinks

Reply:
@RonnyPfannschmidt, well, the whole point of the pull request is not resolving symlinks (I think my comment at https://github.com/pytest-dev/pytest/pull/6523#discussion_r379385235 explains the rationale). 

I'll reword it here, so, hopefully it's clearer: if pytest does resolve symlinks, it's not possible to actually use symlinks to structure projects (be it subst drives, junctions or symlinks) because the paths reported by pytest never match the `cwd` being used for the run and the only usage for symlinks is as a helper to `cd` into a directory (which is what the current symlink resolution accomplishes).

So, what pytest maintainers need to decide is what is the wanted behavior for `pytest` (and I believe those are mutually exclusive, either users are allowed to use symlinks to structure the project or users are allowed to use symlinks as a helper to `cd` into a project).

Reply:
on linux/unix the resolve behaviour makes a lot of sense (and makes things better/avoids certain issues) on windows it seems like the resolve behavour should be to "not" do it if it can be avoided so its a topic where any binary choice is "bad" and a non-binary one may be even worse

in any case its a breaking change that requires a major version, im kinda onboard with making it happen, but i wonder if it should be controllable with a environment variable with sane defaults for unix to resolve and for windows to "not" resolve

Reply:
I'm ok in checking for an env variable and updating the patch (and possibly making the default different for each OS). How do you think it should be named?

Reply:
Not using Linux as a desktop, I can only speak to Windows, but, in my experience, most apps don't do any resolution of symlinks on Windows, so if pytest DOES resolve, paths to the same file will not match. IMHO, an env var is fine for people to configure for those edge cases, but the default for each OS should be the thing that will work for most people. So, in this case, resolve for Linux (as it currently is) and don't resolve for Windows (as it WAS pre-v5.1.0).

Given the above, I would argue that it isn't a breaking change - there is no change on Linux and it is fixing a bug that was introduced in v5.1.0 on Windows. It's probably not that common for people to use subst on Windows so probably not affecting most Windows users either. If there is anything I can do to help beyond giving my opinions, please let me know how. :-)

Reply:
I agree with Adam